test/MinterDeploy.t.sol

```diff
@@ -0,0 +1,187 @@
+
+        bytes memory setMiddlewareData = abi.encodeCall(NetworkMiddlewareService.setMiddleware, (address(hypMinter)));
+        bytes memory networkScheduleData = abi.encodeCall(
+            TimelockController.schedule,
+            (address(networkMiddlewareService), 0, setMiddlewareData, bytes32(0), bytes32(0), 0 days)
+        );
+        bytes memory accessManagerScheduleData = abi.encodeCall(
+            AccessManager.schedule, (SYMBIOTIC_NETWORK, networkScheduleData, uint48(vm.getBlockTimestamp()))
+        );
+
+        // Multisig B can call schedule on network via accessManager
+        vm.prank(multisigB);
+        accessManager.schedule(address(_network), networkScheduleData, uint48(vm.getBlockTimestamp()) + 7 days);
+        skip(7 days);
+        vm.prank(multisigB);
+        accessManager.execute(address(_network), networkScheduleData);
+
+        vm.prank(makeAddr("alice"));
+        _network.execute({
+            target: address(networkMiddlewareService),
+            value: 0,
+            payload: setMiddlewareData,
+            predecessor: bytes32(0),
+            salt: bytes32(0)
+        });
+
+        assertEq(networkMiddlewareService.middleware(SYMBIOTIC_NETWORK), address(hypMinter));
+
+        // Give HypMinter the `MINTER_ROLE` on HYPER contract through the AccessManager
+        // 1. Schedule a grantRole operation via the AccessManager
+        vm.warp(startTimestamp); // Do this the same day as the middleware operations
+        vm.prank(multisigB);
+        bytes memory data = abi.encodeCall(AccessControl.grantRole, (keccak256("MINTER_ROLE"), address(hypMinter)));
+        accessManager.schedule(address(HYPER), data, 0);
+
+        // 2. Execute the operation
+        vm.prank(multisigB);
+        skip(30 days);
+        accessManager.execute(address(HYPER), data);
+
+        console2.log("HypMinter gets MINTER_ROLE at: ", vm.getBlockTimestamp());
+        assertGt(mintDeadline, vm.getBlockTimestamp());
+
+        // Fast forward to after the start time
+        vm.warp(Math.max(vm.getBlockTimestamp(), hypMinter.mintAllowedTimestamp() + 1));
+
+        // Three mints
+        deal(address(HYPER), address(hypMinter), MINT_AMOUNT); // We send the already minted amount to the contract
+        uint256 initialBalance = HYPER.balanceOf(address(REWARDS));
+        hypMinter.mint();
+        hypMinter.mint();
+        hypMinter.mint();
+
+        skip(7 days);
+        uint256 distributionDeadline = 1_761_051_600 + 5 hours; // Tuesday, October 21, 2025 6:00:00 PM GMT (1 week and 5 hours after minting is allowed)
+
+        // Four distributions
+        hypMinter.distributeRewards(firstTimestamp);
+        hypMinter.distributeRewards(firstTimestamp + 30 days);
+        hypMinter.distributeRewards(firstTimestamp + 60 days);
+        hypMinter.distributeRewards(firstTimestamp + 90 days);
+        assertEq(HYPER.balanceOf(address(REWARDS)) - initialBalance, hypMinter.getStakingMintAmount() * 4);
```

Comment:
can you assert that `distributeRewards` reverts after this?

Reply:
Done